iroh/src/tls/verifier.rs

```diff
@@ -126,7 +128,7 @@ impl ServerCertVerifier for ServerCertificateVerifier {
 
                 let end_entity_as_spki = SubjectPublicKeyInfoDer::from(end_entity.as_ref());
 
-                if !self.trusted_spki.contains(&end_entity_as_spki) {
+                if public_key_to_spki(&remote_peer_id) != end_entity_as_spki {
```

Comment:
I wonder if we should instead decode `end_entity` into a `NodeId`

Reply:
I looked into this a bit now, but `SubjectPublicKeyInfoDer` turns out to just be a wrapper around opaque bytes.
`public_key_to_spki` indirectly uses the [spki](https://docs.rs/spki/latest/spki/index.html) crate to encode. I looked at it a bit, but it's... not fun?

FWIW, for Ed25519 keys, there's not much that can go wrong. It's the Ed25519 public key ASN.1 object identifier + the public bytes encoded as bitstring, so effectively a zero byte (indicating there was zero bits added as padding) and the raw public key bytes.

All in all, we *could* match the first 12 bytes of the SPKI against some constant byte string (they should never change), but I'm not sure we should?
Using a library really wouldn't be fun.

Reply:
Now that I think about this, this is effectively doing the same thing: By using `public_key_to_spki` we prefix the expected public key bytes (`remote_peer_id`) with the object identifier and ASN.1 encoding stuff and then verify that the prefix matches as well as the key itself.

:man_shrugging:

Reply:
I guess we can just leave it

Reply:
I'll add a comment with what I learned